Graph conversion

sarangan12 opened the pull request.

veronicagg commented on Gemfile:
```diff
@@ -69,4 +68,5 @@ gem 'azure_mgmt_stream_analytics',          path: 'management/azure_mgmt_stream_
 gem 'azure_mgmt_subscriptions',             path: 'management/azure_mgmt_subscriptions'
 gem 'azure_mgmt_traffic_manager',           path: 'management/azure_mgmt_traffic_manager'
 gem 'azure_mgmt_web',                       path: 'management/azure_mgmt_web'
+gem 'azure_graph',                          path: 'data/azure_graph'
```
should we call this gem graph_rbac? I'm wondering because there was another mention of Graph API (referring to Cosmos DB when we talked about samples) that was different recently.

veronicagg commented on Rakefile:
```diff
@@ -50,6 +50,23 @@ namespace :arm do
       end
     end
+
+    Dir.chdir(File.expand_path('../data', __FILE__))
```
could we create a method we call instead of repeating the code from above and passing "data" folder instead of "management"?

veronicagg commented on Rakefile:
```diff
@@ -70,6 +87,9 @@ namespace :arm do
         elsif(key == 'management')
           Dir.chdir("#{__dir__}/management/#{gem}")
           version = gem_versions['management'][gem]
+        elsif(key == 'data')
```
similar comment as above, could we parameterize this code?

veronicagg commented on Rakefile:
```diff
@@ -286,9 +323,20 @@ def each_gem_dir
     yield subdir
   end
+
+  Dir.chdir("#{__dir__}/data")
```
similar to above comments, could we parameterize this code instead of copy/pasting?

veronicagg commented on Rakefile:
```diff
@@ -300,6 +348,16 @@ def each_child
     end
   end
 
+  # Process Data Gems
```
same here

veronicagg commented on config.json:
```diff
@@ -1,792 +1,791 @@
   "azure_mgmt_analysis_services": {
     "azure_mgmt_analysis_services_2017_07_14": {
-      "markdown": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/7aa3a5247895ba34d6cfec73e036bb66dc907d20/specification/analysisservices/resource-manager/readme.md",
+      "markdown": "https://raw.githubusercontent.com/Azure/azure-rest-api-specs/4a97039db7d2ca3acd5a11fe32d815864bdfc27d/specification/analysisservices/resource-manager/readme.md",
```
are you just updating to the latest commit hash at the time you made this change?

veronicagg commented on data/azure_graph/azure_graph.gemspec:
```diff
@@ -0,0 +1,35 @@
+  spec.version       = Azure::Graph::VERSION
+  spec.authors       = 'Microsoft Corporation'
+  spec.email         = 'azrubyteam@microsoft.com'
+  spec.description   = 'Microsoft Azure Active Directory Graph Data Client Library for Ruby'
```
Not sure we want to say "Data" here, you may just leave it out.

veronicagg commented on data/azure_graph/spec/1.6/graph_spec.rb:
```diff
@@ -0,0 +1,49 @@
+
+require_relative 'spec_helper'
+
+describe 'GraphManagementClient' do
```
left over Management in the describe

veronicagg commented on generators/profilegen/src/defs/definition.rb:
```diff
@@ -0,0 +1,24 @@
+# encoding: utf-8
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
```
could you provide more context/explanation on the files you're adding for profile generation?

veronicagg commented on profile_specs/profiles.json:
```diff
@@ -234,7 +366,9 @@
             "resourceTypes": {
                 "Microsoft.AnalysisServices": {
-                    "2017-07-14": ["*"]
+                    "management": {
```
why do we need a "management" tag here, if we already in the "management" section https://github.com/Azure/azure-sdk-for-ruby/pull/1158/commits/a818cec53fac531fce38474119376a72da2ff63b#diff-a886d4e0a33d2455b5a5344309de4389R363 ?

sarangan12 commented:
Because of the recent merge this PR has got too many conflicts. I have opened a new PR https://github.com/Azure/azure-sdk-for-ruby/pull/1168 for this issue and addressed all your comments there
